java-checks/src/main/java/org/sonar/java/checks/helpers/IdentifierUtils.java

```diff
@@ -35,8 +36,15 @@ private IdentifierUtils() {
     T value = resolver.apply(expression);
     if (value == null && expression.is(Tree.Kind.IDENTIFIER)) {
       ExpressionTree last = ReassignmentFinder.getClosestReassignmentOrDeclarationExpression(expression, ((IdentifierTree) expression).symbol());
-      value = last == null || last == expression ? null : getValue(last, resolver);
+      value = last == null || !isStrictAssignmentOrDeclaration(last) || last == expression ? null : getValue(last, resolver);
```

Comment:
This ternary operator seems barely readable to me. I think I would prefer a plain if/else.